plugins/plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/MachineProviderImplTest.java

```diff
@@ -697,6 +697,56 @@ public void shouldBindCommonAndDevVolumesToContainerOnDevInstanceCreationFromRec
         assertEquals(actualBinds.length, expectedVolumes.size());
         assertEquals(new HashSet<>(asList(actualBinds)), new HashSet<>(expectedVolumes));
     }
+    
+    @Test
+    public void shouldBindVolumesFromToContainerOnDevInstanceCreationFromRecipe() throws Exception {
+    	
+    	List<String> cheServerVolumes = new ArrayList<>(asList("SomeVolumeName:/data/lib"));
+    	String expectedVolumesFrom = "che-server";
+    	
+    	provider = new MachineProviderBuilder().setVolumesFrom(expectedVolumesFrom)
+    									       .build();
+    	
+    	final boolean isDev = true;
+    	
+    	CheServiceImpl service = createService();
+    	service.setContainerName(expectedVolumesFrom);
+    	service.setVolumes(cheServerVolumes);
+    	createInstanceFromRecipe(isDev, service);
+    	
+        ArgumentCaptor<CreateContainerParams> argumentCaptor = ArgumentCaptor.forClass(CreateContainerParams.class);
+        verify(dockerConnector).createContainer(argumentCaptor.capture());
+        verify(dockerConnector).startContainer(any(StartContainerParams.class));
+        
+        final String[] actualVolumesFrom = argumentCaptor.getValue().getContainerConfig().getHostConfig().getVolumesFrom();
+        assertTrue(Arrays.asList(actualVolumesFrom).contains(expectedVolumesFrom));
+    }
+    
+    @Test
+    public void shouldBindVolumesFromToContainerOnDevInstanceCreationFromRecipeWhenServiceHasVolumesFromSet() throws Exception {
+    	
+    	List<String> cheServerVolumes = new ArrayList<>(asList("SomeVolumeName:/data/lib"));
+    	List<String> cheServerVolumesFrom = new ArrayList<>(asList("SomeContainerName"));
+    	String expectedVolumesFrom = "che-server";
+    	
+    	provider = new MachineProviderBuilder().setVolumesFrom(expectedVolumesFrom)
+    									       .build();
+    	
+    	final boolean isDev = true;
+    	
+    	CheServiceImpl service = createService();
+    	service.setContainerName(expectedVolumesFrom);
+    	service.setVolumes(cheServerVolumes);
+    	service.setVolumesFrom(cheServerVolumesFrom);
+    	createInstanceFromRecipe(isDev, service);
+    	
+        ArgumentCaptor<CreateContainerParams> argumentCaptor = ArgumentCaptor.forClass(CreateContainerParams.class);
+        verify(dockerConnector).createContainer(argumentCaptor.capture());
+        verify(dockerConnector).startContainer(any(StartContainerParams.class));
+        
+        final String[] actualVolumesFrom = argumentCaptor.getValue().getContainerConfig().getHostConfig().getVolumesFrom();
+        assertTrue(Arrays.asList(actualVolumesFrom).contains(expectedVolumesFrom));
```

Comment:
If one **volumesFrom** is provided in **MachineProviderImpl** and another in **CheServiceImpl** then looks like **.getHostConfig().getVolumesFrom()** should return array of length 2. But **expectedVolumesFrom** doesn't contain **expectedVolumesFrom**. Am I missing something?

BTW Testng has **assertEqualsNoOrder** for comparing arrays.